pkg/math/oracle/median.go

```diff
@@ -33,279 +41,170 @@ func NewMedianAggregator(logger *zap.Logger, cfg mmtypes.MarketMap) (*MedianAggr
-//  3. BTC/USD = BTC/USD
-//
-// The final median price for BTC/USD will be the median of the prices calculated from the above
-// calculations.
-func (m *MedianAggregator) AggregateFn() types.PriceAggregationFn {
-	return func(feedsPerProvider types.AggregatedProviderPrices) types.TickerPrices {
-		// Calculate the median price for each price feed.
-		feedMedians := median.ComputeMedian()(feedsPerProvider)
-		m.logger.Info("calculated median prices for raw price feeds", zap.Int("num_prices", len(feedMedians)))
-
-		for ticker, price := range feedMedians {
-			m.logger.Info(
-				"got median price",
-				zap.String("ticker", ticker.String()),
-				zap.String("price", price.String()),
-			)
-		}
-
-		// Calculate the converted prices for each ticker using the conversion markets.
-		aggregatedMedians := m.GetConvertedPrices(feedMedians)
-		m.logger.Info("calculated median prices for converted price feeds", zap.Int("num_prices", len(aggregatedMedians)))
-
-		// Replace the median prices for each ticker with the aggregated median prices.
-		// This will overwrite the median prices for each ticker with the final aggregated median prices.
-		// This is necessary because the conversion markets may not be provided for all tickers.
-		for ticker, price := range aggregatedMedians {
-			m.logger.Info(
-				"replacing median price",
-				zap.String("ticker", ticker.String()),
-				zap.String("price", price.String()),
-			)
-
-			feedMedians[ticker] = price
-		}
-
-		return feedMedians
-	}
-}
-
-// GetConvertedPrices returns the converted prices for each ticker using the provided median
-// prices and the conversion markets.
-func (m *MedianAggregator) GetConvertedPrices(feedMedians types.TickerPrices) types.TickerPrices {
-	// Scale all of the medians to a common number of decimals. This does not lose precision.
-	scaledMedians := make(types.TickerPrices)
-	for ticker, price := range feedMedians {
-		scaledPrice, err := ScaleUpCurrencyPairPrice(ticker.Decimals, price)
-		if err != nil {
-			m.logger.Error(
-				"failed to scale price",
-				zap.Error(err),
-				zap.String("ticker", ticker.String()),
-				zap.Uint64("decimals", ticker.Decimals),
-				zap.String("price", price.String()),
-			)
-
-			continue
-		}
-
-		scaledMedians[ticker] = scaledPrice
-	}
-
-	// Determine the final aggregated price for each ticker, specifically only for the tickers
-	// that have a set of conversion markets.
-	aggregatedMedians := make(types.TickerPrices)
-	for tickerStr, paths := range m.cfg.Paths {
-		ticker, ok := m.cfg.Tickers[tickerStr]
+// The index price cache contains the previously calculated median prices.
+func (m *MedianAggregator) AggregatedData() {
+	updatedPrices := make(types.TickerPrices)
+	for ticker, paths := range m.cfg.Paths {
+		target, ok := m.cfg.Tickers[ticker]
 		if !ok {
 			m.logger.Error(
-				"failed to get ticker",
-				zap.String("ticker", tickerStr),
+				"failed to get ticker; skipping aggregation",
+				zap.String("ticker", ticker),
```

Comment:
Let's cover this case

Reply:
This implies an invalid config, which is going to come in a follow up PR.